pilot/pkg/features/pilot.go

```diff
@@ -120,6 +120,14 @@ var (
 	WorkloadEntryCrossCluster = env.Register("PILOT_ENABLE_CROSS_CLUSTER_WORKLOAD_ENTRY", true,
 		"If enabled, pilot will read WorkloadEntry from other clusters, selectable by Services in that cluster.").Get()
 
+	// For Gateway SDS, the 'config cluster' is already the default - no change.
+	RemoteClusterAccess = env.Register("PILOT_REMOTE_CLUSTER_ACCESS", true,
```

Comment:
This env name is a little big, i was thinking the intention is only to limit access of the secret.  

But actually here you limited every access to other cluster

Reply:
Yes. It is true by default - so no backward incompatible changes. 

If you install Istio per cluster - there is no good reason to allow other clusters to connect to it and grab the config. If one cluster is compromised ( in the 'istiod per cluster' mode) - it is useful to reduce the blast radius and not let the attacker get the configs or do a DOS on other clusters. Secret is the obviously huge hole - but not the only one.

Reply:
It could also be deploying central istiod in one cluster, all the workload proxies connect to it from other clusters. As the central istiod cluster is managed, all the configs, like secrets could be created only in the config cluster.

Reply:
That's probably the best way to run Istio - but it's not the most common, and we can't set the defaults for the common use case ( which is one istiod per cluster ) for 'central istiod' mode. 

Users of central istiod will need to turn the flag over.